ocaml/xapi/xapi_xenops.ml

```diff
@@ -2349,8 +2349,19 @@ let events_from_xapi () =
                 while true do
                   let api_timeout = 60. in
                   let timeout = 30. +. api_timeout +. !Db_globs.master_connection_reset_timeout in
-                  let otherwise () = info "Event.from timed out in %f seconds, abort the events listening loop and retry" timeout; raise Exit in
-                  let from = Helpers.timebox ~timeout ~otherwise (fun () -> XenAPI.Event.from ~rpc ~session_id ~classes ~token:!token ~timeout:api_timeout |> event_from_of_rpc) in
+                  let timebox_rpc = Helpers.make_timeboxed_rpc ~__context timeout in
```

Comment:
This sits inside a `Helpers.call_api_functions` where we also make a connection, get an `rpc` function and do a login. We need the session, but it seems that we now have two connections, while we really need to cut down on slave->master connections.

Reply:
Not quite: we only actually make a connection _during_ an RPC call: until the `rpc` parameter is supplied to `make_rpc` it's only partially applied. So I don't believe we're creating more connections with this change.

Reply:
Oh, I see! I didn't realise that this was partially applied.